integrations/acquisition/covidcast/test_is_latest_issue.py

```diff
@@ -0,0 +1,346 @@
+    rows = [
+      CovidcastRow('src', 'sig', 'day', 'state', 20200414, 'pa', 
+                   1.5, 2.5, 3.5, nmv, nmv, nmv, 20200414, 0),
+      CovidcastRow('src', 'sig', 'day', 'state', 20200415, 'wa', 
+                   3, 2, 1, nmv, nmv, nmv, 20200415, 0)
+    ]
+    self._db.insert_or_update_bulk(rows)
+    self._db.run_dbjobs()
+    #preview
+    self._db._cursor.execute('''SELECT * FROM `signal_history`''')
+    self.totalRows = len(list(self._db._cursor.fetchall()))
+    self._db._cursor.execute('''SELECT * FROM `signal_latest`''')
+    record = self._db._cursor.fetchall()
+    x = PrettyTable()
+    x.field_names = ['signal_data_id',
+    'signal_key_id',
+    'geo_key_id',
+    'demog_key_id' ,
+    'issue' ,
+    'data_as_of_dt',
+    'time_type',
+    'time_value' ,
+    'reference_dt' ,
+    'value' ,
+    'stderr' ,
+    'sample_size',
+    'lag' ,
+    'value_updated_timestamp' ,
+    'computation_as_of_dt']
+    print("SIGNAL_LATEST TABLE")
+    for row in record:
+      x.add_row(list(row)[:len(x.field_names)])
+    print(x)
+    print("Finish with 1st Set of Data")
+
+    #sanity check for adding dummy data
+    sql = '''SELECT `issue` FROM `signal_latest` where `time_value` = 20200414'''
+    self._db._cursor.execute(sql)
+    record = self._db._cursor.fetchall()
+    self.assertEqual(record[0][0], 20200414)
+
+    #when uploading data patches (data in signal load has < issue than data in signal_latest)
+      #INSERT OLDER issue (does not end up in latest table)
+      #INSERT NEWER issue (ends up in latest table)
+    #when signal_load is older than signal_latest, we patch old data (i.e changed some old entries)
+    newRow = [
+    CovidcastRow('src', 'sig', 'day', 'state', 20200414, 'pa', 
+                  4.4, 4.4, 4.4, nmv, nmv, nmv, 20200416, 2)] #should show up
+    self._db.insert_or_update_bulk(newRow)
+    self._db.run_dbjobs()
+    
+    #check newer issue in signal_latest
+    sql = '''SELECT `issue` FROM `signal_latest` where `time_value` = 20200414 '''
+    self._db._cursor.execute(sql)
+    record = self._db._cursor.fetchall()
+    self.assertEqual(record[0][0], 20200416) #new data added
+
+    updateRow = [
+      CovidcastRow('src', 'sig', 'day', 'state', 20200414, 'pa', 
+                  6.5, 2.2, 11.5, nmv, nmv, nmv, 20200414, 2)]  #should not showup
+    self._db.insert_or_update_bulk(updateRow)
+    self._db.run_dbjobs()
+    
+    #check newer issue in signal_latest
+    sql = '''SELECT `issue` FROM `signal_latest` where `time_value` = 20200414  '''
+    self._db._cursor.execute(sql)
+    record2 = self._db._cursor.fetchall()
+    self.assertEqual(record, record2) #same as previous as is_latest did not change
+
+    #dynamic check
+    sql2 = '''SELECT `issue` FROM `signal_history` where `time_value` '''
+    self._db._cursor.execute(sql2)
+    record3 = self._db._cursor.fetchall()
+    self.assertEqual(3,self.totalRows + 1) #ensure 3 added (1 of which refreshed)
```

Comment:
i dont think this is checking what you want...  `totalRows` doesnt get updated and is still equal to 2, which you set at the very top of this test

Reply:
Here was what I was trying to test: Initially, there are 2 rows and I update one of them and added a new one, so I wanted to make sure that there were only 3 rows in the end. If this is confusing, I am trying to think of better ways to test it.
(1 old & not updated)
(1 old but updated)
(1 added newly)